stock_partner_delivery_window/models/res_partner.py

```diff
@@ -180,3 +183,76 @@ def copy_data(self, default=None):
         ]
         result["delivery_time_window_ids"] = [(0, 0, val) for val in values]
         return [result]
+
+    def next_delivery_window_start_datetime(self, from_date=None, timedelta_days=None):
+        """Get next starting datetime in a preferred delivery window.
+
+        If from_date is already in a delivery window, from_date is "the next"
+
+        :param from_date: Datetime object (Leave empty to use now())
+        :param timedelta_days: Number of days to use in the computation
+                               (Leave empty to use 7 days or 1 week)
+        :return: Datetime object
+        """
+        self.ensure_one()
+        if not from_date:
+            from_date = datetime.now()
+        if self.is_in_delivery_window(from_date):
+            return fields.Datetime.to_datetime(from_date)
+        if timedelta_days is None:
+            timedelta_days = 7
```

Comment:
```suggestion
        timedelta_days = timedelta_days or 7
```
unless you support `0`.

Reply:
IMO it's usually safer to consider 0 when working on dates/times and deltas